mwdb/core/search/exceptions.py

```diff
@@ -1,29 +1,54 @@
-class SQLQueryBuilderBaseException(Exception):
+from typing import Optional, Tuple, Type
+
+from luqum.tree import Item
+
+
+class QueryBaseException(Exception):
     """
     Base exception for SQLQueryBuilder
     """
 
 
-class UnsupportedGrammarException(SQLQueryBuilderBaseException):
+class QueryParseException(QueryBaseException):
     """
-    Raised when SQLQueryBuilder does not support given Lucene grammar
+    Raised when Lucene parser is unable to parse a query
     """
 
 
-class FieldNotQueryableException(SQLQueryBuilderBaseException):
+class UnsupportedNodeException(QueryBaseException):
+    def __init__(self, message: str, node: Item):
+        super().__init__(f"{message} ({node.pos}:{node.pos + node.size - 1})")
+
+
+class UnsupportedNodeTypeException(UnsupportedNodeException):
     """
-    Raised when field does not exists, so it can't be queried, eg. file.unexistent_field
+    Raised when query visitor does not support given Lucene grammar
     """
 
+    def __init__(self, node: Item, expected: Optional[Tuple[Type, ...]] = None):
```

Comment:
```suggestion
    def __init__(self, node: Item, expected: Optional[Tuple[Type, ...]] = None) -> None:
```
black will complain here